routers/api/v1/notify/user.go

```diff
@@ -81,7 +81,7 @@ func ListNotifications(ctx *context.APIContext) {
 		return
 	}
 	err = nl.LoadAttributes()
-	if err != nil {
+	if err != nil && !models.IsErrCommentNotExist(err) {
```

Comment:
But it may panic between a nil comment in the template file?

Reply:
As I'm quite new to the codebase, I'm not really sure, what template you mean?

However, looking through what I *think* it does, it seems to [handle such cases properly](https://github.com/go-gitea/gitea/blob/062fd4c217cc7302f56acf043d6214a9db46ee2f/modules/convert/notification.go#L40-L43). Also, when I tested this, a notification which previously caused a `500 Internal Server Error` just returned an empty string as `latest_comment_url` and `latest_comment_html_url`, which is the normal behaviour in this case (I assume).

Reply:
I don't think any template is of relevance here, this is a API route, not the web route.

@eladyn The templates are the "html" files but in [golang's syntax](https://golangdocs.com/templates-in-golang) so we can have Server-Side-Rendering.

Reply:
In fact, this is a bug of the function `LoadAttributes`, we just need to fix this function to make `err` a local variable.
```go
func (nl NotificationList) LoadAttributes() (err error) {
	for i := 0; i < len(nl); i++ {
		err = nl[i].LoadAttributes()
		if err != nil && !IsErrCommentNotExist(err) {
			return
		}
	}
	return
}
```

Reply:
Right, thanks for the hint! I fixed this and also removed the check in `.../repo.go` since that would then be duplicate.